pkg/pubsub/adapter/converters/storage_test.go

```diff
@@ -19,171 +19,151 @@ package converters
 import (
 	"context"
 	"testing"
+	"time"
 
 	"cloud.google.com/go/pubsub"
-	"github.com/google/go-cmp/cmp"
+	"github.com/google/knative-gcp/pkg/apis/events/v1beta1"
 
-	cloudevents "github.com/cloudevents/sdk-go"
-	cepubsub "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub"
-	pubsubcontext "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub/context"
-	"github.com/google/knative-gcp/pkg/apis/events/v1alpha1"
+	cev2 "github.com/cloudevents/sdk-go/v2"
+)
+
+const (
+	bucket    = "my-bucket"
+	objectId  = "myfile.jpg"
+	eventType = "OBJECT_FINALIZE"
+)
+
+var (
+	storagePublishTime = time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC)
 )
 
 func TestConvertCloudStorageSource(t *testing.T) {
 
 	tests := []struct {
 		name        string
-		message     *cepubsub.Message
-		sendMode    ModeType
-		wantEventFn func() *cloudevents.Event
+		message     *pubsub.Message
+		wantEventFn func() *cev2.Event
 		wantErr     bool
 	}{{
 		name: "no attributes",
-		message: &cepubsub.Message{
+		message: &pubsub.Message{
 			Data: []byte("test data"),
-			Attributes: map[string]string{
-				"knative-gcp": "com.google.cloud.storage",
-			},
 		},
-		sendMode: Binary,
-		wantEventFn: func() *cloudevents.Event {
-			return storageCloudEvent(map[string]string{})
+		wantEventFn: func() *cev2.Event {
+			return storageCloudEvent()
 		},
 		wantErr: true,
 	}, {
 		name: "no bucketId attribute",
-		message: &cepubsub.Message{
+		message: &pubsub.Message{
 			Data: []byte("test data"),
 			Attributes: map[string]string{
-				"knative-gcp": "com.google.cloud.storage",
-				"eventType":   "OBJECT_FINALIZE",
-				"attribute1":  "value1",
-				"attribute2":  "value2",
-			},
-		},
-		sendMode: Binary,
-		wantEventFn: func() *cloudevents.Event {
-			return storageCloudEvent(map[string]string{
+				"eventType":  eventType,
 				"attribute1": "value1",
 				"attribute2": "value2",
-			})
+			},
+		},
+		wantEventFn: func() *cev2.Event {
+			return storageCloudEvent()
 		},
 		wantErr: true,
 	}, {
 		name: "no eventType attribute",
-		message: &cepubsub.Message{
+		message: &pubsub.Message{
 			Data: []byte("test data"),
 			Attributes: map[string]string{
-				"knative-gcp": "com.google.cloud.storage",
-				"bucketId":    "my-bucket",
+				"bucketId": bucket,
+				"objectId": objectId,
 			},
 		},
-		sendMode: Binary,
-		wantEventFn: func() *cloudevents.Event {
-			return storageCloudEvent(map[string]string{})
+		wantEventFn: func() *cev2.Event {
```

Comment:
it is used here?

Reply:
it's set here but I don't see where it's used

Reply:
actually this should be removed, you are right... 
doing it in a follow up